### ISSUE TITLE
Explore leveraging OpenAPI core data model functionality of apicurio-data-models

### ISSUE THREAD

**Issue by EricWittmann** [smallrye/smallrye-open-api#276]
As initially suggested in this comment:  https://github.com/smallrye/smallrye-open-api/pull/275#issuecomment-600152810

I wonder if perhaps we can leverage the mature reading/writing/validation of OpenAPI content (with support for both 2.0 and 3.0) built in the https://github.com/Apicurio/apicurio-data-models repository.  The models found in that library may not be compatible with the interfaces defined by the Microprofile OpenAPI spec, but it's probably worth exploring.  The apicurio-data-models library provides some useful functionality beyond simple reading/writing of the content, including:

* Validation
* Transformation from OpenAPI 2.0 to OpenAPI 3.0
* Visitor and Traverser patterns

**Comment by MikeEdgar** [comment]
@EricWittmann , @phillip-kruger - I took a quick look at this and one of things that might be a challenge is incorporating the MicroProfile model's interfaces. Eric, I recall you mentioning that very point when we discussed the topic last year. 

Since a user of the `OASModelReader` or `OASFilter` could theoretically provide an implementation for one of the interfaces that wasn't created from the `OASFactory`, we will probably still need at least a writer that is based purely on the interfaces.

Is it worth exploring the possibility of having Apicurio data models themselves support the model interfaces of MicroProfile directly? Some of that might come in handy for any future Async API specification as well, assuming the "schema" parts align with OpenAPI.
